From 35947aba37adb4f352735212f6dfab39ede0cb54 Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Sun, 18 Jan 2026 22:10:24 +0100
Subject: [PATCH] fix: Improve test reliability on loaded build systems

Reduce CPU crunch time in testMultithreadedAlg to avoid timeouts and
increase sleep margins in BaseSinkUnitTest autoflush test to handle
thread scheduling delays.
---
 GaudiHive/options/testMultithreadedAlg.py  | 4 ++--
 GaudiKernel/tests/src/BaseSinkUnitTest.cpp | 6 +++---
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/GaudiHive/options/testMultithreadedAlg.py b/GaudiHive/options/testMultithreadedAlg.py
index b0f3c797f..67365f0ec 100644
--- a/GaudiHive/options/testMultithreadedAlg.py
+++ b/GaudiHive/options/testMultithreadedAlg.py
@@ -1,6 +1,6 @@
 #!/usr/bin/env gaudirun.py
 #####################################################################################
-# (c) Copyright 1998-2019 CERN for the benefit of the LHCb and ATLAS collaborations #
+# (c) Copyright 1998-2026 CERN for the benefit of the LHCb and ATLAS collaborations #
 #                                                                                   #
 # This software is distributed under the terms of the Apache version 2 licence,     #
 # copied verbatim in the file "LICENSE".                                            #
@@ -44,7 +44,7 @@ CPUCrunchSvc(shortCalib=True)
 
 a1 = CPUCruncher("A1")
 a1.Cardinality = cardinality
-a1.avgRuntime = 5.0
+a1.avgRuntime = 0.5
 a1.NParallel = 5
 a1.OutputLevel = DEBUG
 
diff --git a/GaudiKernel/tests/src/BaseSinkUnitTest.cpp b/GaudiKernel/tests/src/BaseSinkUnitTest.cpp
index 903d799fd..d6b150c19 100644
--- a/GaudiKernel/tests/src/BaseSinkUnitTest.cpp
+++ b/GaudiKernel/tests/src/BaseSinkUnitTest.cpp
@@ -1,5 +1,5 @@
 /***********************************************************************************\
-* (c) Copyright 1998-2021 CERN for the benefit of the LHCb and ATLAS collaborations *
+* (c) Copyright 1998-2026 CERN for the benefit of the LHCb and ATLAS collaborations *
 *                                                                                   *
 * This software is distributed under the terms of the Apache version 2 licence,     *
 * copied verbatim in the file "LICENSE".                                            *
@@ -85,9 +85,9 @@ BOOST_AUTO_TEST_CASE( test_sink_autoflush ) {
     TestFlushSink     sink{ output, app.as<ISvcLocator>() };
     sink.setProperty( "AutoFlushPeriod", .5 ).ignore();
     sink.start().ignore();
-    std::this_thread::sleep_for( 600ms );
+    std::this_thread::sleep_for( 1000ms );
     BOOST_TEST( output.str() == RegularLine );
-    std::this_thread::sleep_for( 600ms );
+    std::this_thread::sleep_for( 1000ms );
     BOOST_TEST( output.str() == RegularLine + RegularLine );
     sink.stop().ignore();
     BOOST_TEST( output.str() == RegularLine + RegularLine + FinalLine );
-- 
2.48.1

